### ISSUE TITLE
[question] export()+source() as scm attribute replacement

### ISSUE THREAD

**Issue by tbsuht** [conan-io/conan#13894]
### What is your question?

Hi,

I'm currently replacing the scm feature in my recipes as proposed in the migration guide [here](https://docs.conan.io/1/migrating_to_2.0/recipes.html#the-scm-attribute).

What I noticed is that when building in our CI there is a difference in the approaches which I'm not sure can be achieved with the new export()+source() approach.

In our CI pipeline we checkout a git repo and then call multiple `conan create` commands. With the scm feature I will notice something like this in the logs:
```
foobar/1.2.0: Repo origin deduced by 'auto': ssh://git@bitbucket.foobar.com:1234/foobar/foobar.git
foobar/1.2.0: Revision deduced by 'auto': 9970eea7b07868444beb73e09ce471903be03b92
foobar/1.2.0: SCM: Getting sources from folder: /home/jenkins/workspace/foobar_foobar/src
```

So the already available checkout is used.

Now with export()+source() we will effectively do two checkouts: one to get the conanfile and then another inside the conan cache.

Is there a way to reuse the already available sources? I don't want to store them inside the package.

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @tbsuht 

Thanks for your question.

> Is there a way to reuse the already available sources? I don't want to store them inside the package.

With then new export()+source() approach, the sources are not stored inside the package. I am not sure what you mean, but only the url and commit will be stored in the ``conandata.yml``, and then the ``source()`` method does a local checkout, but those sources are not stored in the package, never uploaded to the server, that only happens locally when building from source, but only exists in that local cache, not in the package. Could you please clarify a bit?

**Comment by tbsuht** [comment]
Sorry, this was misleading.

with export()+source() a code change in a git repo will have this effect in our CI:
trigger jenkins pipeline -> checkout repo on the node -> call conan create -> checkout repo in the cache

With the scm feature I noticed that the second checkout in the cache is not taking place and the checkout on the node is used/copied. Is this also possible with the new approach?

What I meant with storing the sources inside package: an alternative would be to use export/export_sources to get rid of the additional checkout but I don't want to store the sources inside the package.

**Comment by memsharded** [comment]
I see

> With the scm feature I noticed that the second checkout in the cache is not taking place and the checkout on the node is used/copied. Is this also possible with the new approach?

No, this was exactly one of the most flawed things that we decided to remove. It had some important bugs and design flaws. We decided that it was much better to have an extra checkout than doing such a copy from a user folder. Using the ``fetch_commit()`` it can be done more efficiently, and in any case, as this only happens when building from source that typically takes a lot of time, the time of the extra checkout is just a very small fraction of the time, not really impacting the overall time.
